bftengine/include/bftengine/ReplicaConfig.hpp

```diff
@@ -198,10 +198,17 @@ class ReplicaConfig : public concord::serialize::SerializableFactory<ReplicaConf
 
   CONFIG_PARAM(prePrepareFinalizeAsyncEnabled, bool, true, "Enabling asynchronous preprepare finishing");
 
-  CONFIG_PARAM(threadbagConcurrency,
-               uint32_t,
-               64u,
-               "Number of threads given to thread pool that is created for any request processing");
+  CONFIG_PARAM(
+      threadbagConcurrencyLevel1,
+      uint32_t,
+      48u,
+      "Number of threads given to thread pool that is created for any request processing for the parent of validation");
+
+  CONFIG_PARAM(
+      threadbagConcurrencyLevel2,
+      uint32_t,
+      32u,
```

Comment:
We are requested to minimize the threads' usage. Are you sure about those numbers?

Reply:
Thanks @yuliasherman for the comment. The total number of thread are increasing from 64 to 80. This is a big increase, so I think I should keep the total thread count same. The outer thread will be used by all the messages, so I will keep that at 40. The inner thread is only for PrePrepareMsg, so I will keep that at 24. 
The result for threadcount of threadpool >= 32 showed performance improvement as per our experiments. So Keeping outer thread to 40 will not have any performance degradation.